packages/patternfly-4/react-table/src/components/Table/Body.js

```diff
@@ -9,69 +9,80 @@ const propTypes = {
   /** Specify key which should be used for labeling each row. */
   rowKey: PropTypes.string,
   /** Function that is fired when user clicks on row.  */
-  onRowClick: PropTypes.func
+  onRowClick: PropTypes.func,
+  /** Virtualized rows (optional provided in place of rows) */
+  rowsToRender: PropTypes.array
 };
 
 const defaultProps = {
   rowKey: 'id',
   className: '',
-  onRowClick: () => undefined
+  onRowClick: () => undefined,
+  rowsToRender: undefined
 };
 
 class ContextBody extends React.Component {
   onRow = (row, props) => {
-    const { onRowClick } = this.props;
-    return ({
+    const { onRowClick, onRow } = this.props;
+
+    // if we have a parent onRow, call it to retrieve the resulting props
+    const parentRowProps = onRow ? onRow(row, props) : {};
+    return {
       isExpanded: row.isExpanded,
       isOpen: row.isOpen,
-      onClick: (event) => onRowClick(event, row, props)
-    });
-  }
+      onClick: event => onRowClick(event, row, props),
+      ...parentRowProps
+    };
+  };
 
   parentsExpanded(parentId) {
     const { rows } = this.props;
-    return rows[parentId].hasOwnProperty('parent') ? this.parentsExpanded(rows[parentId].parent) : rows[parentId].isOpen;
+    return rows[parentId].hasOwnProperty('parent')
+      ? this.parentsExpanded(rows[parentId].parent)
+      : rows[parentId].isOpen;
   }
 
   render() {
-    const { className, headerData, rows, rowKey, children, onRowClick, ...props } = this.props;
+    const { className, headerData, rows, rowKey, rowsToRender, children, onRowClick, ...props } = this.props;
     let shiftKey = 0;
     shiftKey += headerData[0] && headerData[0].extraParams.onSelect ? 1 : 0;
     shiftKey += headerData[0] && headerData[0].extraParams.onCollapse ? 1 : 0;
-    const mappedRows = headerData.length !== 0 && rows.map((oneRow, oneRowKey) => {
-      return {
+    const mappedRows =
+      headerData.length !== 0 &&
+      rows.map((oneRow, oneRowKey) => ({
         ...oneRow,
-        ...oneRow && (oneRow.cells || oneRow).reduce(
-          (acc, curr, key) => {
-            return ({
+        ...(oneRow &&
+          (oneRow.cells || oneRow).reduce(
+            (acc, curr, key) => ({
               ...acc,
               ...{
                 [headerData[shiftKey + key].property]: {
                   title: curr.title || curr,
                   props: curr.props
                 }
               }
-            })
-          },
-          { id: oneRowKey }),
-        ...oneRow.parent ? {
-          isExpanded: this.parentsExpanded(oneRow.parent) && rows[oneRow.parent].isOpen
-        } : {},
-      }
-    });
+            }),
+            { id: oneRow[rowKey] || oneRowKey }
```

Comment:
we need a specific `row-id` (row key) rendered on the table rows currently viewed to support virtualized scroll (a simple counter index would not support this).

Reply:
i think this is good to allow consumer to override the `id` column if desired